Dev

EdinsonRequena opened the pull request.

EdinsonRequena committed:
Add new files and update pylint configuration

EdinsonRequena committed:
Add Stock.csv file with initial data

pep8speaks commented:
Hello @EdinsonRequena! Thanks for updating this PR. We checked the lines you've touched for [PEP 8](https://www.python.org/dev/peps/pep-0008) issues, and found:





* In the file [`app/database/manager.py`](https://github.com/EdinsonRequena/data_optimization_test/blob/8aa2c252d9f064387b2b85703babec205bad43e6/app/database/manager.py):

> [Line 25:80](https://github.com/EdinsonRequena/data_optimization_test/blob/8aa2c252d9f064387b2b85703babec205bad43e6/app/database/manager.py#L25): [E501](https://duckduckgo.com/?q=pep8%20E501) line too long (81 > 79 characters)
> [Line 26:80](https://github.com/EdinsonRequena/data_optimization_test/blob/8aa2c252d9f064387b2b85703babec205bad43e6/app/database/manager.py#L26): [E501](https://duckduckgo.com/?q=pep8%20E501) line too long (113 > 79 characters)
> [Line 53:80](https://github.com/EdinsonRequena/data_optimization_test/blob/8aa2c252d9f064387b2b85703babec205bad43e6/app/database/manager.py#L53): [E501](https://duckduckgo.com/?q=pep8%20E501) line too long (86 > 79 characters)
> [Line 81:80](https://github.com/EdinsonRequena/data_optimization_test/blob/8aa2c252d9f064387b2b85703babec205bad43e6/app/database/manager.py#L81): [E501](https://duckduckgo.com/?q=pep8%20E501) line too long (82 > 79 characters)
> [Line 91:80](https://github.com/EdinsonRequena/data_optimization_test/blob/8aa2c252d9f064387b2b85703babec205bad43e6/app/database/manager.py#L91): [E501](https://duckduckgo.com/?q=pep8%20E501) line too long (84 > 79 characters)

* In the file [`app/database/test_manager.py`](https://github.com/EdinsonRequena/data_optimization_test/blob/8aa2c252d9f064387b2b85703babec205bad43e6/app/database/test_manager.py):

> [Line 30:80](https://github.com/EdinsonRequena/data_optimization_test/blob/8aa2c252d9f064387b2b85703babec205bad43e6/app/database/test_manager.py#L30): [E501](https://duckduckgo.com/?q=pep8%20E501) line too long (91 > 79 characters)

* In the file [`app/utils/logger_config.py`](https://github.com/EdinsonRequena/data_optimization_test/blob/8aa2c252d9f064387b2b85703babec205bad43e6/app/utils/logger_config.py):

> [Line 14:80](https://github.com/EdinsonRequena/data_optimization_test/blob/8aa2c252d9f064387b2b85703babec205bad43e6/app/utils/logger_config.py#L14): [E501](https://duckduckgo.com/?q=pep8%20E501) line too long (86 > 79 characters)





##### Comment last updated at 2024-01-20 08:01:42 UTC

EdinsonRequena committed:
Add test cases for read_csv function

EdinsonRequena committed:
Add unit tests for DatabaseManager class

EdinsonRequena committed:
Add PostgreSQL service for CI testing

EdinsonRequena committed:
Update PostgreSQL version and add health checks for the service

EdinsonRequena committed:
Add PostgreSQL availability check and test coverage configuration

EdinsonRequena committed:
Add DB_HOST and DB_PORT environment variables

EdinsonRequena committed:
Update database configuration

EdinsonRequena committed:
Commented out PostgreSQL availability check in CI workflow

EdinsonRequena committed:
Update PostgreSQL configuration in CI workflow

EdinsonRequena committed:
Commented out test coverage step in CI workflow

EdinsonRequena committed:
Update Python CI workflow

This commit updates the Python CI workflow by adding a step to install the PostgreSQL client and check the PostgreSQL service before running tests with coverage.

EdinsonRequena committed:
Update database environment variables in CI workflow

EdinsonRequena committed:
Update database environment variables

EdinsonRequena committed:
Remove test database configuration and test execution

EdinsonRequena committed:
Update PostgreSQL service command to include username

EdinsonRequena committed:
Commented out test coverage and SonarCloud scan actions
